catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -226,25 +319,49 @@ private String getRegistryId(Metacard mcard) {
                 .toString();
     }
 
-    public void init() throws FederationAdminException {
-
-        List<Metacard> registryMetacards = federationAdmin.getRegistryMetacards();
-
+    public void init()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+        List<Metacard> registryMetacards;
+        Filter registryFilter = filterBuilder.attribute(Metacard.TAGS)
+                .is()
+                .like()
+                .text(RegistryConstants.REGISTRY_TAG);
+        Subject systemSubject = Security.getInstance()
+                .getSystemSubject();
+        QueryImpl query = new QueryImpl(registryFilter);
+        query.setPageSize(1000);
+        QueryRequest request = new QueryRequestImpl(query);
+        request.getProperties()
+                .put(SecurityConstants.SECURITY_SUBJECT, systemSubject);
+
```

Comment:
Since this is a catalog plugin it sees all registry metacards that go through the framework so after initialization it keeps track of all the registry-id via the post ingest methods.